util/kubernetes.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"regexp"
+)
+
+var K8sSelectorRegexp = regexp.MustCompile(`\s*([^!=\s]+)\s*([!=]+)\s*([^\s]+)\s*`)
+
+// CheckLabelSelectorMismatch checks if selectors do not match the given labels
+func CheckLabelSelectorMismatch(labels map[string]string, selectors []string) bool {
```

Comment:
Is this actually called anywhere (other than the tests)? I didn't see any call sites.

It might also be clearer to invert the condition here (i.e., `CheckLabelSelectorMatch` or `CheckLabelSelectorMatches`) and then negate it in the caller.

Separately, since we're dealing with multiple selectors and multiple labels, it might be good to include the full expected behavior (i.e., if I'm reading this correctly, that every selector must match its corresponding label, if any) in the comment.

Reply:
> Is this actually called anywhere (other than the tests)? I didn't see any call sites.

No, `CheckLabelSelectorMismatch` is not called (though `K8sSelectorRegexp` is called, and that's the reason behind of making this util). I'd like to use the k8s code for this and refactor, so that I can retire this whole util. As this was a good chunk, I decided to keep them together (also makes easier to write tests focusing on this).

> It might also be clearer to invert the condition here (i.e., `CheckLabelSelectorMatch` or `CheckLabelSelectorMatches` or maybe just `SelectorsMatchLabels`) and then negate it in the caller.

I thought about it, but the nice part of the current logic is that it can early return easily when something doesn't match (also logic is a bit simpler). Well, given neither labels and selectors won't be that long, maybe I don't need to worry about too much for the performance?

> Separately, since we're dealing with multiple selectors and multiple labels, it might be good to include the full expected behavior (i.e., if I'm reading this correctly, that every selector must match its corresponding label, if any) in the comment.

I'll point to k8s label selector docs (that I linked in the other comment) 👍 FYI, the short answer to this is the following:

> In the case of multiple requirements, all must be satisfied so the comma separator acts as a logical AND (&&) operator.